### PR TITLE
A0-788: Add descriptive errors to proposal validation

### PR DESCRIPTION
**Pull request by obrok**
# Description

Adds descriptive errors to proposal validation.
Closes https://cardinal-cryptography.atlassian.net/browse/A0-788

## Type of change

- Refactoring (non-breaking change)

# Checklist:

- I have added tests
- I have created new documentation

### CODE REVIEW

File: finality-aleph/src/data_io/proposal.rs [hunk 1]
```diff
@@ -29,6 +38,15 @@ pub struct UnvalidatedAlephProposal<B: BlockT> {
     pub number: NumberFor<B>,
 }
 
+/// Represents possible invalid states as described in [UnvalidatedAlephProposal].
+#[derive(Debug, Display, PartialEq, Error)]
+pub enum ValidationError {
+    BranchTooLarge,
```
Comment (piotrMocz) [inline finality-aleph/src/data_io/proposal.rs:44]: Just a suggestion: we could consider adding some numbers to the fields here to make it slightly more meaningful, e.g. the actual branch length and the limit here. Not necessary though, thinking out loud ;)

**Review by piotrMocz — APPROVED** [review]
Looks very good, just one suggestion :+1:

File: finality-aleph/src/data_io/proposal.rs (hunk 1 above)
Comment (obrok) [reply finality-aleph/src/data_io/proposal.rs:44]: Yeah, was thinking about it. Guess I'll do it after all.

Comment (obrok) [reply finality-aleph/src/data_io/proposal.rs]: :heavy_check_mark: Added details to the errors

**Review by DamianStraszak — APPROVED** [review]
Good job!

**Comment by obrok** [comment]
:heavy_check_mark: @DamianStraszak addressed your points

**Pull request merged**